packages/front-end/services/track.ts

```diff
@@ -77,3 +78,25 @@ export default function track(
 
   jitsu.track(event, trackProps);
 }
+
+export function trackSnapshot(
+  event: "create" | "update" | "delete" | "error",
+  props: {
+    source: string;
+    experiment: string;
+    engine: StatsEngine;
+    regressionAdjustmentEnabled: boolean;
+    sequentialTestingEnabled: boolean;
+    sequentialTestingTuningParameter?: number;
+    skipPartialData: boolean;
+    activationMetricSelected: boolean;
+    queryFilterSelected: boolean;
+    segmentSelected: boolean;
+    dimension: string;
+    error?: string;
+  }
+): void {
+  props.experiment = md5(props.experiment);
+  props.dimension = props.dimension ? md5(props.dimension) : "";
```

Comment:
The dimension has 2 parts - the type of dimension and the specific dimension id.

It's currently a string in one of the following formats:
- `pre:date`
- `pre:activation`
- `exp:(.*)`
- `(.*)`

The only part we need to obfuscate is the `(.*)` above.  I think we should parse this string and store it in 2 separate columns.  Will make the data much easier to query and analyze.

```sql
WHERE dimension_type = 'predefined' AND dimension_id = 'date'
```

Reply:
done using `substring` rather than some kind of split, because users can have `:` in the dim name and that makes knowing if it is our colon or theirs that we are running in to.